frontend/packages/date-time/src/group.ts

```
@@ -0,0 +1,82 @@
+ * @returns - 겹치는 날짜 범위를 그룹화하여 반환
+ */
+export const groupByOverlap = <T extends DateRangeWithId>(dateInfos: T[]): OverlapDate<T>[] => 
+  dateInfos.reduce((groups, current) => {
+    const lastGroup = groups[groups.length - 1];
+    const isNewGroup = 
+      !lastGroup || lastGroup[lastGroup.length - 1].end <= current.start;
+
+    if (isNewGroup) return [...groups, [{ ...current, idx: 0 }]];
+
+    return [
+      ...groups.slice(0, -1), 
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)